### PR TITLE
Build: Update baseline gradle plugin to 5.58.0

### PR DESCRIPTION
**Pull request by findepi**
Among other things, this pulls newest version of error-prone which adds
new useful checks.

### CODE REVIEW

**Comment by findepi** [comment]
Draft -- currently depends on https://github.com/apache/iceberg/pull/10518

**Review by jbonofre — COMMENTED** [review]
I guess this PR "conflicts" with the JDK8 drop one right ?

**Comment by findepi** [comment]
i think it depends on the other. we can update only after we drop 8.

**Comment by Fokko** [comment]
@findepi time to revisit this? :)

**Comment by findepi** [comment]
Yes! let me rebase

File: baseline.gradle [hunk 1]
```diff
@@ -86,11 +86,17 @@ subprojects {
           '-Xep:CollectionUndefinedEquality:ERROR',
           // specific to Palantir - Uses name `log` but we use name `LOG`
           '-Xep:ConsistentLoggerName:OFF',
+          // TODO this is a recently added check. Figure out whether we adjust the code or suppress for good
```
Comment (nastra) [inline baseline.gradle:89]: I believe we should probably address all of these and then configure them to `ERROR`.
Do you have a rough idea of how big the diff would be when addressing all of the new error prone checks? I think if it's not too much then we can include it in this PR.
Alternatively, we can open up separate issues and mark them as `good first issue` to give other people a chance to contribute here

Comment (Fokko) [reply baseline.gradle:89]: > Alternatively, we can open up separate issues and mark them as good first issue to give other people a chance to contribute here

I like that 👍

Comment (findepi) [reply baseline.gradle]: Created https://github.com/apache/iceberg/issues/10853, https://github.com/apache/iceberg/issues/10854, https://github.com/apache/iceberg/issues/10855

**Review by nastra — APPROVED** [review]

**Pull request merged**